File: backend/providers/stepfun.js
module.exports = {

  name: 'stepfun',
  aliases: ['step', 'step-ai'],
  displayName: 'StepFun (阶跃星辰)',
  description: 'StepFun API — Step-1 and Step-2 models',
  baseUrl: 'https://api.stepfun.com/v1',
  apiMode: 'openai_chat',
  envVars: ['STEPFUN_API_KEY', 'STEPFUN_BASE_URL'],
  authType: 'api_key',
  defaultModel: 'step-2-16k',
  fallbackModels: ['step-1-128k', 'step-1-32k'],
  defaultMaxTokens: 4096,
  signupUrl: 'https://platform.stepfun.com',
  supportsHealthCheck: true,
  modelProfiles: {
    'step-2-16k': { supportsReasoning: false, supportsThinking: false, combinedBudget: false, contextWindow: 16384, maxOutputTokens: 4096 },
    'step-1-8k': { supportsReasoning: false, supportsThinking: false, combinedBudget: false, contextWindow: 8192, maxOutputTokens: 4096 },
    'step-1-32k': { supportsReasoning: false, supportsThinking: false, combinedBudget: false, contextWindow: 32768, maxOutputTokens: 4096 },
    'step-1-128k': { supportsReasoning: false, supportsThinking: false, combinedBudget: false, contextWindow: 131072, maxOutputTokens: 4096 },
    'step-1v-32k': { supportsReasoning: false, supportsThinking: false, combinedBudget: false, contextWindow: 32768, maxOutputTokens: 4096 },
    '*': { supportsReasoning: false, supportsThinking: false, combinedBudget: false, contextWindow: 131072, maxOutputTokens: 4096 },
  },

};

File: backend-py/app/providers/stepfun.py
"""
Provider config for StepFun (阶跃星辰).
"""
from __future__ import annotations
from typing import Any
INFO: dict[str, Any] = {'name': 'StepFun (阶跃星辰)', 'aliases': ['step', 'step-ai'], 'display_name': 'StepFun (阶跃星辰)', 'description': 'StepFun API — Step-1 and Step-2 models', 'base_url': 'https://api.stepfun.com/v1', 'api_mode': 'openai_chat', 'env_vars': ['STEPFUN_API_KEY', 'STEPFUN_BASE_URL'], 'auth_type': 'api_key', 'default_model': 'step-2-16k', 'fallback_models': ['step-1-128k', 'step-1-32k'], 'default_max_tokens': 4096, 'signup_url': 'https://platform.stepfun.com', 'supports_health_check': True, 'model_profiles': {'step-2-16k': {'supportsReasoning': False, 'supportsThinking': False, 'combinedBudget': False, 'contextWindow': 16384, 'maxOutputTokens': 4096}, 'step-1-8k': {'supportsReasoning': False, 'supportsThinking': False, 'combinedBudget': False, 'contextWindow': 8192, 'maxOutputTokens': 4096}, 'step-1-32k': {'supportsReasoning': False, 'supportsThinking': False, 'combinedBudget': False, 'contextWindow': 32768, 'maxOutputTokens': 4096}, 'step-1-128k': {'supportsReasoning': False, 'supportsThinking': False, 'combinedBudget': False, 'contextWindow': 131072, 'maxOutputTokens': 4096}, 'step-1v-32k': {'supportsReasoning': False, 'supportsThinking': False, 'combinedBudget': False, 'contextWindow': 32768, 'maxOutputTokens': 4096}, '*': {'supportsReasoning': False, 'supportsThinking': False, 'combinedBudget': False, 'contextWindow': 131072, 'maxOutputTokens': 4096}}}

def resolveBaseUrl() -> str:
    return 'https://api.stepfun.com/v1'

def resolveApiKey(envKey: str | None=None) -> str | None:
    return envKey